src/components/crossDevice/MobileFlow.js

```diff
@@ -36,21 +36,19 @@ class MobileFlow extends Component {
 
   onClientSuccess = () => {
     this.props.actions.setClientSuccess(true)
-    this.props.actions.mobileConnected(false)
   }
 
   onDisconnectPing = (data) => {
     this.sendMessage('disconnect pong', data.roomId)
-    this.props.actions.mobileConnected(false)
   }
 
   render = (props) => {
     if (this.props.clientSuccess)
       return <CrossDeviceSubmit {...props}/>
     if (this.props.mobileConnected)
       return <MobileConnected {...props}/>
-
-    return <MobileNotificationSent {...props}/>
+    if (!this.props.mobileConnected)
+      return <MobileNotificationSent {...props}/>
```

Comment:
How about:
`return this.props.mobileConnected ? <MobileConnected {...props}/> : <MobileNotificationSent {...props}/>`